hs_core/receivers.py

```diff
@@ -141,3 +146,22 @@ def metadata_element_pre_update_handler(sender, **kwargs):
         return {'is_valid': True, 'element_data_dict': element_form.cleaned_data}
     else:
         return {'is_valid': False, 'element_data_dict': None, "errors": element_form.errors}
+
+
+@receiver(post_add_generic_aggregation)
+@receiver(post_add_geofeature_aggregation)
+@receiver(post_add_raster_aggregation)
+@receiver(post_add_netcdf_aggregation)
+@receiver(post_add_timeseries_aggregation)
+@receiver(post_add_reftimeseries_aggregation)
+@receiver(post_remove_file_aggregation)
+@receiver(post_delete_resource)
+@receiver(post_raccess_change)
+def hs_update_web_services(sender, **kwargs):
+    """Signal to update resource web services."""
+    update_web_services(
+        services_url=hs_settings.HSWS_URL,
+        api_token=hs_settings.HSWS_API_TOKEN,
+        timeout=hs_settings.HSWS_TIMEOUT,
+        res_id=kwargs.get("resource").short_id
+    )
```

Comment:
@kjlippold You are directly calling update_web_services() as a function which will run synchronously. To execute it asynchronously as a celery task, you will have to call apply_async() to the function task. Search apply_async in hydroshare code base to see how it is used.

Reply:
I just noticed this a little bit ago and fixed it, so it'll be changed in the next commit.